spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -163,6 +151,36 @@ public Publisher createPublisher(String topic) {
 		});
 	}
 
+	void applyPublisherSettings(Publisher.Builder publisherBuilder) {
+		if (this.executorProvider != null) {
+			publisherBuilder.setExecutorProvider(this.executorProvider);
+		}
+
+		if (this.channelProvider != null) {
+			publisherBuilder.setChannelProvider(this.channelProvider);
+		}
+
+		if (this.credentialsProvider != null) {
+			publisherBuilder.setCredentialsProvider(this.credentialsProvider);
+		}
+
+		if (this.headerProvider != null) {
+			publisherBuilder.setHeaderProvider(this.headerProvider);
+		}
+
+		if (this.retrySettings != null) {
+			publisherBuilder.setRetrySettings(this.retrySettings);
+		}
+
+		if (this.batchingSettings != null) {
+			publisherBuilder.setBatchingSettings(this.batchingSettings);
+		}
+
+		if (this.enableMessageOrdering != null) {
```

Comment:
If it's primitive, then we don't need to check for null here, just straight set the value, or, to be extra safe, check if true.